### DISCUSSION TITLE
documentation for German dependency label inventory

### DISCUSSION THREAD

**Discussion by josefkr** [explosion/spaCy#12558] [Language Support]
The documentation for the dependency labels used by the German parser gives the labels and with `spacy.explain` one can see a disabbreviated form of the labels. 

But I didn't manage to find a reference to annotation guidelines or links to papers or treebanks that can be consulted on what the labels are supposed to mean. 

As far as I can tell the current inventory of 42 labels is not identical with either Tiger or NEGRA. For instance, spacy's labels for German include `avc` which is in NEGRA but not in Tiger. The label `ag` is used by spacy. It is not in NEGRA but it is in Tiger. spacy's inventory also includes `dep` which is neither in NEGRA nor in Tiger. 

Could you steer me towards documentation of the label set that spaCy uses?

## Which page or section is this issue related to?
https://spacy.io/models/de

**Comment by adrianeboyd** [comment]
The dependency labels other than `dep` come directly from the sources TIGER and Tiger2Dep. I only see `avc` in one example in the official TIGER Annotationsschema publication, but it's included in the TIGER Introduction and in the TIGER XML header as an edge label ("adverbial phrase component") and I see ~100 occurrences in TIGER v2 (dec05). The TIGER documentation is currently available here: https://www.ims.uni-stuttgart.de/forschung/ressourcen/korpora/tiger/

`dep` is a placeholder label used by spacy in some cases where no other label fits, usually only for whitespace or as a backoff label in the dependency parser for a few rare cases. It's basically the same as Universal Dependencies `dep`: https://universaldependencies.org/u/dep/dep.html

**Comment by josefkr** [comment]
Thank you for this info. I think it would be great if  the table that lists the labels had information on the labels and/or training data (or a link to some place where that is given)

![Selection_023](https://user-images.githubusercontent.com/9550305/233626156-7cba669d-1b53-412b-9c60-39c13a9bd64c.png)

Actually, looking now at the pdf  on Tiger that IMS hosts, https://www.ims.uni-stuttgart.de/documents/ressourcen/korpora/tiger-corpus/annotation/tiger_scheme-syntax.pdf , I find that it lists oa2, which is not among the labels that spacy lists. Did Tiger2Dep merge oa2 with oa or is that something that spaCy did internally? (I quickly grepped thru Tiger2Dep's (v1.3) files and couldn't find anything that looks like oa2 gets merged into oa there).

**Reply by adrianeboyd** [reply]
Due to our TIGER license we can't currently publish the internal training data, so we can only refer you to the sources listed on the models page in the "Sources" section under e.g. https://spacy.io/models/de#de_core_news_sm. The same information is included in the model packages in `meta.json` and with license information in `LICENSES_SOURCES`. We typically don't have any more documentation than is provided by the original sources.

Some of this predates my time working on spacy, but I think that the main internal additions that have been made to our TIGER training corpora beyond what's available in the published tiger2dep conversion are 1) an alignment to the raw texts (with spaces) and 2) a conversion from the original pos/morph annotation to UD UPOS/FEATS, which is based on the tigerdep2ud conversion tool mentioned on the UD tools page (https://universaldependencies.org/tools#truducer) but which is unfortunately no longer online.

`oa2` is in the training data but the parser uses some frequency thresholds that mean that it is not included as a label for this particular trained model (related to `min_action_freq` in the parser config). I think the output would probably be `oa` or possibly `sb` for these cases in practice, but I haven't done a lot of testing. Also be aware that rules in the `attribute_ruler` component in the trained pipelines can potentially modify the final dependency parse output. For German, I think this just affects the output for whitespace tokens, but the exact set of rules varies by language and pipeline.

In general the provided trained pipelines are intended to be a useful starting point, but if you need more fine-grained control of the training data or settings then we'd recommend developing a custom pipeline from scratch. For a corpus in CoNLL-U format, the [`pipelines/tagger_parser_ud`](https://github.com/explosion/projects/tree/v3/pipelines/tagger_parser_ud) project could provide a good starting point.

**Comment by josefkr** [comment]
Many thanks for all that info! That certainly accounts for the  labels that do or don't occur in the output.
